logging/measure/tracer.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package measure
+
+//go:generate protoc --proto_path=model/apitrace --gogo_out=model/apitrace model/apitrace/apitrace.proto
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/logging/measure/model/apitrace"
+)
+
+// Tracer allows to measure, store and list measured time entries.
+type Tracer interface {
+	// LogTime puts measured time to the table and resets the time.
+	LogTime(entity string, start time.Time)
+	// Get all trace entries stored
+	Get() *apitrace.Trace
+	// Clear removes entries from the log database
+	Clear()
+}
+
+// NewTracer creates new tracer object
+func NewTracer(name string, log logging.Logger) Tracer {
+	return &tracer{
+		name:   name,
+		log:    log,
+		index:  1,
+		timedb: make([]*entry, 0),
+	}
+}
+
+// Inner structure handling database and measure results
+type tracer struct {
+	sync.Mutex
+
+	name string
+	log  logging.Logger
+	// Entry index, used in database as key and increased after every entry. Never resets since the tracer object is
+	// created or the database is cleared
+	index uint32
```

Comment:
This better be `uint64` as I suggested [before](https://github.com/ligato/cn-infra/pull/340/#discussion_r218747362). We don't currently have any limit and clear doesn't reset this index so we want to go for largest possible integer. Change the type of index in the model too.